docs/design/query-caching.md

```diff
@@ -0,0 +1,224 @@
+
+> A shared cache MUST NOT use a cached response to a request with an Authorization header field (Section 11.6.2 of [HTTP]) to satisfy any subsequent request unless the response contains a Cache-Control field with a response directive (Section 5.2.2) that allows it to be stored by a shared cache, and the cache conforms to the requirements of that directive for that response. [Storing Responses to Authenticated Requests](https://www.rfc-editor.org/rfc/rfc9111#name-storing-responses-to-authen)
+
+Due to how cache keys and values are created in the scope of this document, caching Data API builder responses in CDNs and browsers is not in scope. This means that `cache-control` **response** directives are not in scope. While the spec does state that the server can return `cache-control` if desired, DAB performs authentication and authorization on requests prior to accessing the in-memory cache. That same level of validation can't occur in the browser or CDN headers for security purposes. We can validate that ASP.NET core returns `private` or `no-store` as the `cache-control` header value.
+
+- Honor HTTP Header `cache-control` per https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control with the following two **request** directives:
+  1. `no-cache` - Get a fresh response from the database and updates DAB's cache with the fresh result.
+  1. `no-store` - Do not cache the request or response. Do not attempt to fetch a result from the cache.
+
+Note that `cache-control` **request** directives will be ignored by browser and CDN caches when an `Authorization` header is included in the request. 
+
+Mozilla docs https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#preventing_storing suggest that "the most restrictive directive should be honored" when directives conflict. When both `no-cache` and `no-store` are present, `no-store` wins.
+
+## Database object compatibility
+
+| Database Objec Type | Cache compatible?                                            |
+|---------------------|--------------------------------------------------------------|
+| Table               | Yes                                                          |
+| View                | Yes                                                          |
+| Stored-Procedure    | Yes, only for stored procedures which query data. See below. |
+
+Stored procure entities must have the following configuration to allow for response caching.
+
+```json
+{
+  "entityName": {
+    "rest": {
+      "methods": [
```

Comment:
So, by default data returned by stored procs is NOT cache compatible unless they change the methods/operation properties. 
Also, what happens if REST method is GET but graphql operation is not modified to query and vice versa..

Reply:
right.

so if there are conflicting REST: GET and GraphQL: Mutation for a stored proc,then the cache service may need to know the context of the request and check config accordingly: graphql mutation -> skip cache.